index.js

```diff
@@ -3,6 +3,8 @@
 const PluginVirtual = require('./src/lib/plugin')
 const Token = require('./src/util/token')
 
-module.exports = PluginVirtual
+module.exports = PluginVirtual.bind(null, {})
+
+module.exports.MakePluginVirtual = (channel) => (PluginVirtual.bind(null, channel))
```

Comment:
A class is just a function (which is it's constructor). MakePluginVirtual returns a bound version of the PluginVirtual constructor.